sdk/test/metadata.test.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { Metadata } from '../src/metadata';
+
+describe('Metadata', () => {
+  it('should return empty array for missing keys', () => {
+    // Arrange
+    const meta = new Metadata();
+
+    // Act
+    const res = meta.get('something');
+
+    // Assert
+    expect(res.length).to.be.equal(0);
+  });
+
+  it('should return data that is set', () => {
+    // Arrange
+    const meta = new Metadata();
+    meta.set('key1', 'hello');
+
+    // Act
+    const res = meta.get('key1');
+
+    // Assert
+    expect(res.length).to.be.equal(1);
+    expect(res[0]).to.be.equal('hello');
+  });
+
+  it('should return multiple data that if set', () => {
+    // Arrange
+    const meta = new Metadata();
+    meta.set('key1', 'hello1');
+    meta.set('key1', 'hello2');
+
+    // Act
+    const res = meta.get('key1');
+
+    // Assert
+    expect(res.length).to.be.equal(2);
+    expect(res[0]).to.be.equal('hello1');
+    expect(res[1]).to.be.equal('hello2');
```

Comment:
Expect that chai matchers can also match the whole array to check same length and members? This also looks fine.